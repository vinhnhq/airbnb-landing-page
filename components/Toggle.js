import React, { Fragment } from "react";
import PropTypes from "prop-types";

import styles from "../styles/Toggle.module.scss";

export default function Toggle({ isOn, handleToggle, onColor }) {
  return (
    <Fragment>
      <input
        id="toggle-checkbox-id"
        type="checkbox"
        checked={isOn}
        onChange={handleToggle}
        className={styles.input}
      />
      <label
        htmlFor="toggle-checkbox-id"
        style={{ background: isOn && onColor }}
        className={styles.label}
      >
        <span className={styles.button} />
      </label>
    </Fragment>
  );
}

Toggle.propTypes = {
  isOn: PropTypes.bool.isRequired,
  handleToggle: PropTypes.func.isRequired,
  onColor: PropTypes.string,
};
