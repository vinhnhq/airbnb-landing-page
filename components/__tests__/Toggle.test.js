import React from "react";
import renderer from "react-test-renderer";

import Toggle from "../Toggle";

describe("components --> Toggle", () => {
  const mockProps = {
    isOn: true,
    handleToggle: () => {},
    onColor: "red",
  };

  test("should render correctly", () => {
    const tree = renderer.create(<Toggle {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
