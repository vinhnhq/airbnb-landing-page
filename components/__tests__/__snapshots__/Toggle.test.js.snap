// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components --> Toggle should render correctly 1`] = `
Array [
  <input
    checked={true}
    className="input"
    id="toggle-checkbox-id"
    onChange={[Function]}
    type="checkbox"
  />,
  <label
    className="label"
    htmlFor="toggle-checkbox-id"
    style={
      Object {
        "background": "red",
      }
    }
  >
    <span
      className="button"
    />
  </label>,
]
`;
